### ISSUE TITLE
StackExchange.Redis.RedisConnectionException: It was not possible to connect to the redis server(s)

### ISSUE THREAD

**Issue by reneklacan** [StackExchange/StackExchange.Redis#578]
Sorry for creating this kind of issue again but I went through all related issue and I couldn't find anything that would help.

My environment:
```
OS: macOS 10.12.4
Mono: 4.6.2
StackExchange.Redis: 1.2.1
Language: F#
Redis Server: 3.2.8
```

My code:
```fsharp
let sw = new StringWriter()
let redis = ConnectionMultiplexer.Connect("127.0.0.1:6379", sw)
```

Error:
```
StackExchange.Redis.RedisConnectionException: It was not possible to connect to the redis server(s); to create a disconnected multiplexer, disable AbortOnConnectFail.
  at StackExchange.Redis.ConnectionMultiplexer.ConnectImpl (System.Func`1[TResult] multiplexerFactory, System.IO.TextWriter log) [0x0006e] in <0df32117c5364b2ba164cc500243df89>:0
  at StackExchange.Redis.ConnectionMultiplexer.Connect (System.String configuration, System.IO.TextWriter log) [0x00005] in <0df32117c5364b2ba164cc500243df89>:0
  at <StartupCode$FSI_0025>.$FSI_0025.main@ () [0x0000b] in <9761f99abd9a464baed36ad80013783f>:0
  at (wrapper managed-to-native) System.Reflection.MonoMethod:InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00038] in <8f2c484307284b51944a1a13a14c0266>:0
Stopped due to error
```

Logs:
```
Connecting 127.0.0.1:6379/Interactive...
BeginConnect: 127.0.0.1:6379
1 unique nodes specified
Requesting tie-break from 127.0.0.1:6379 > __Booksleeve_TieBreak...
Allowing endpoints 00:00:05 to respond...
Awaiting task completion, IOCP: (Busy=0,Free=800,Min=8,Max=800), WORKER: (Busy=3,Free=797,Min=8,Max=800)
EndConnect: 127.0.0.1:6379
Connected Interactive/127.0.0.1:6379
Server handshake
Auto-configure...
Sending critical tracer: Interactive/127.0.0.1:6379
Writing to Interactive/127.0.0.1:6379: ECHO
Flushing outbound buffer
Starting poll
Connect complete: 127.0.0.1:6379
Not all tasks completed cleanly, IOCP: (Busy=0,Free=800,Min=8,Max=800), WORKER: (Busy=6,Free=794,Min=8,Max=800)
127.0.0.1:6379 did not respond
Awaiting task completion, IOCP: (Busy=0,Free=800,Min=8,Max=800), WORKER: (Busy=6,Free=794,Min=8,Max=800)
Not all tasks completed cleanly, IOCP: (Busy=0,Free=800,Min=8,Max=800), WORKER: (Busy=6,Free=794,Min=8,Max=800)
127.0.0.1:6379 failed to nominate (WaitingForActivation)
No masters detected
127.0.0.1:6379: Standalone v2.0.0, master; keep-alive: 00:01:00; int: ConnectedEstablishing; sub: ConnectedEstablishing; not in use: DidNotRespond
127.0.0.1:6379: int ops=6, qu=2, qs=6, qc=0, wr=0, socks=1; sub ops=2, qu=0, qs=2, qc=0, wr=0, socks=1
Circular op-count snapshot; int: 0+6=6 (0.60 ops/s; spans 10s); sub: 0+2=2 (0.20 ops/s; spans 10s)
Sync timeouts: 0; fire and forget: 0; last heartbeat: -1s ago
resetting failing connections to retry...
retrying; attempts left: 2...
1 unique nodes specified
Requesting tie-break from 127.0.0.1:6379 > __Booksleeve_TieBreak...
Allowing endpoints 00:00:05 to respond...
Awaiting task completion, IOCP: (Busy=0,Free=800,Min=8,Max=800), WORKER: (Busy=6,Free=794,Min=8,Max=800)
Not all tasks completed cleanly, IOCP: (Busy=0,Free=800,Min=8,Max=800), WORKER: (Busy=6,Free=794,Min=8,Max=800)
127.0.0.1:6379 did not respond
Awaiting task completion, IOCP: (Busy=0,Free=800,Min=8,Max=800), WORKER: (Busy=6,Free=794,Min=8,Max=800)
Not all tasks completed cleanly, IOCP: (Busy=0,Free=800,Min=8,Max=800), WORKER: (Busy=6,Free=794,Min=8,Max=800)
127.0.0.1:6379 failed to nominate (WaitingForActivation)
No masters detected
127.0.0.1:6379: Standalone v2.0.0, master; keep-alive: 00:01:00; int: ConnectedEstablishing; sub: ConnectedEstablishing; not in use: DidNotRespond
127.0.0.1:6379: int ops=12, qu=2, qs=6, qc=0, wr=0, sync=6, async=2, socks=2; sub ops=4, qu=0, qs=2, qc=0, wr=0, sync=2, socks=2
Circular op-count snapshot; int: 0+12=12 (1.20 ops/s; spans 10s); sub: 0+4=4 (0.40 ops/s; spans 10s)
Sync timeouts: 0; fire and forget: 0; last heartbeat: -1s ago
resetting failing connections to retry...
retrying; attempts left: 1...
1 unique nodes specified
Requesting tie-break from 127.0.0.1:6379 > __Booksleeve_TieBreak...
Allowing endpoints 00:00:05 to respond...
Awaiting task completion, IOCP: (Busy=0,Free=800,Min=8,Max=800), WORKER: (Busy=6,Free=794,Min=8,Max=800)
Not all tasks completed cleanly, IOCP: (Busy=0,Free=800,Min=8,Max=800), WORKER: (Busy=6,Free=794,Min=8,Max=800)
127.0.0.1:6379 did not respond
Awaiting task completion, IOCP: (Busy=0,Free=800,Min=8,Max=800), WORKER: (Busy=6,Free=794,Min=8,Max=800)
Not all tasks completed cleanly, IOCP: (Busy=0,Free=800,Min=8,Max=800), WORKER: (Busy=6,Free=794,Min=8,Max=800)
127.0.0.1:6379 failed to nominate (WaitingForActivation)
No masters detected
127.0.0.1:6379: Standalone v2.0.0, master; keep-alive: 00:01:00; int: ConnectedEstablishing; sub: ConnectedEstablishing; not in use: DidNotRespond
127.0.0.1:6379: int ops=18, qu=2, qs=6, qc=0, wr=0, sync=12, async=4, socks=3; sub ops=6, qu=1, qs=2, qc=0, wr=0, sync=4, socks=3
Circular op-count snapshot; int: 0+18=18 (1.80 ops/s; spans 10s); sub: 0+6=6 (0.60 ops/s; spans 10s)
Sync timeouts: 0; fire and forget: 0; last heartbeat: -1s ago
```

I am 100% sure that Redis server is ok and accessible. I've tested Ruby, Elixir and Python clients and `redis-cli` command line tool as well and all of them are connecting successfully.

I've also tried `"localhost"`, `"localhost:6379"`, `"127.0.0.1"` as argument but same result.

I tried https://github.com/ctstone/csredis and it works but I need pub/sub support and CSRedis doesn't support it.

Any idea what's up?

**Comment by reneklacan** [comment]
I ended up using https://github.com/andrew-bn/RedisBoost which works for me but this should probably be addressed anyway.

**Comment by Rombersoft** [comment]
I have some error

**Comment by hesampour** [comment]
I have the same problem. any suggestion

**Comment by reneklacan** [comment]
@hesampour Have you tried to use https://github.com/andrew-bn/RedisBoost I shared above? It worked well for me without any problems.

**Comment by Rombersoft** [comment]
I also use it

**Comment by hesampour** [comment]
thanks for your answer. I run the code. the problem was that I have not started the redis server at all. we should start the redis server at first. when we install the package we should go to package folder and run rhe redis server.
I should test the redis boost too.

**Comment by hesampour** [comment]
now I have another problem. I am implementing an asp api for an angular 4 project. and I want to check If a person has attempted 3 false login and the block it. with this package If I set the login link to 3 times it counts all tries even if it was true case and user name and pasword is correct. but it is not the main problem the main problem is that it blocks all the links

**Comment by benzsuankularb** [comment]
Also got this. It's very severe.
Only with .netframework (mono) but work on netcore

**Comment by hesampour** [comment]
@benzsuankularb you need to run the redis server from package folder. go to packages/tools and run the redis-server.exe

**Comment by NickCraver** [comment]
A lot of timeout w.r.t threading issues were resolved with the 1.2.6 build due to a config change there. Given we have a few issues tracking this and a rewrite of the socket code is on our radar, I'm going to collapse out the duplicates here. However, I've tagged them and will update these issues when we've got something to test on the new-network-layer side.
